core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -189,6 +189,7 @@ final class ShuffleBlockFetcherIterator(
         // Filter out zero-sized blocks
         localBlocks ++= blockInfos.filter(_._2 != 0).map(_._1)
         numBlocksToFetch += localBlocks.size
+        shuffleMetrics.incLocalBytesRead(blockInfos.map(_._2).sum)
```

Comment:
Is there a reason you are not incrementing the localBytesRead in fetchLocalBlocks right after the localBlocks are incremented?
Looking at the initialization code, i don't really think there is much difference in functionality but it might be easier to understand to have both:
shuffleMetrics.incLocalBlocksFetched(1)
shuffleMetrics.incLocalBytesRead(buf.size)
be together. Also, if they are together there is less chance of them later being calculated at different times. (if we want to add more parallelism for example).